dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -917,6 +919,105 @@ public User setUserPrivilege(@Parameter(hidden = true, name = "user")@Auth User
+        cloudInstanceToBeAdded.setUrl(cloudInstanceBody.getUrl());
+        cloudInstanceToBeAdded.setSupportsFileImports(cloudInstanceBody.isSupportsFileImports());
+        cloudInstanceToBeAdded.setSupportsHttpImports(cloudInstanceBody.isSupportsHttpImports());
+        cloudInstanceToBeAdded.setSupportedLanguages(cloudInstanceBody.getSupportedLanguages());
+        // TODO: Figure how to make this not required (already adding the instance to the user)
+        cloudInstanceToBeAdded.setUser(user);
+
+        user.getCloudInstances().add(cloudInstanceToBeAdded);
+        return user.getCloudInstances();
+    }
+
+    @DELETE
+    @Timed
+    @UnitOfWork()
+    @Path("/{userId}/cloudInstances/{cloudInstanceId}")
+    @Operation(operationId = "deleteUserCloudInstance", description = "Delete a cloud instances belonging to the user", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = HttpStatus.SC_OK + ""),
+            @ApiResponse(responseCode = HttpStatus.SC_FORBIDDEN + ""),
+            @ApiResponse(responseCode = HttpStatus.SC_UNAUTHORIZED + ""),
+            @ApiResponse(responseCode = HttpStatus.SC_NOT_FOUND + "")
+    })
+    @ApiOperation(value = "See OpenApi for details", hidden = true)
+    public void deleteUserCloudInstance(@Parameter(hidden = true, name = "user")@Auth User authUser,
+            @PathParam("userId") @Parameter(name = "userId", in = ParameterIn.PATH, description = "User to delete cloud instance for", required = true) long userId,
+            @PathParam("cloudInstanceId") @Parameter(name = "cloudInstanceId", in = ParameterIn.PATH, description = "Cloud instance Id to remove from user", required = true) long cloudInstanceId) {
+        final User user = userDAO.findById(userId);
+        checkUser(authUser, userId);
+        boolean deleted = user.getCloudInstances().removeIf(cloudInstance -> cloudInstance.getId() == cloudInstanceId);
+        if (!deleted) {
+            throw new CustomWebApplicationException("ID of cloud instance does not exist", HttpStatus.SC_NOT_FOUND);
+        }
+    }
+
+    @PUT
+    @Timed
+    @UnitOfWork()
+    @Path("/{userId}/cloudInstances/{cloudInstanceId}")
+    @Operation(operationId = "putUserCloudInstance", description = "Delete a cloud instances belonging to the user", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    @Consumes(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "See OpenApi for details", hidden = true)
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = HttpStatus.SC_OK + ""),
+            @ApiResponse(responseCode = HttpStatus.SC_FORBIDDEN + ""),
+            @ApiResponse(responseCode = HttpStatus.SC_UNAUTHORIZED + ""),
+            @ApiResponse(responseCode = HttpStatus.SC_NOT_FOUND + "")
+    })
+    public void putUserCloudInstance(@Parameter(hidden = true, name = "user")@Auth User authUser,
```

Comment:
I think it should return the created CloudInstance

Reply:
Will change on frontend developer demand. Until then, leaving as is.  For reference when i was working on this:

- https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT
- https://stackoverflow.com/a/827045
- https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html